### PR TITLE
Coinacceptor client device implementation.

### PR DESCRIPTION
**Pull request by plaes**
Here comes the Coin Acceptor client device implementation.

Also bundles some general changes to existing APIs:
* Replaced `serial` with `serialport` library and also reworked the `SerialClient` / `CCTalkDevice` inititalizations a bit.
* Adds simple `cctalk-host` and `coinacceptor` examples. Nothing fancy for now, but at least shows how to use them :)
* Also comes with tests (for coin acceptor client device). These are still hacky and could be simplified, but at least work.

Future TODO list:
* Make the CoinAcceptor embeddable in multi-threaded application (our initial approach is will be probably event polling using `mpsc`. Basically we have to listen for master inhibit and send out coin in events).

And some ideas that are currently out of scope, but worth thinking about:
* It would be also nice to drop the current hard dependency on serialport and instead refactor the whole serialport to use generic interface types....
* ...which would be a first step towards eventual `no-std` support (yay for rust-embedded).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,5 +8,16 @@ description = "CCTalk protocol implementation for handling payment devices"
...
+[[example]]
+name = "coinacceptor"
+
+[[example]]
+name = "cctalk-host"
+
+[dev-dependencies]
+env_logger = "0.7"
+clap = "2.33.3"
+serialport = "4.0.1"
```
Comment (blackghost1987) [inline Cargo.toml]: why is this needed here? serialport is also in base dependencies

**Review by blackghost1987 — CHANGES_REQUESTED** [review]
Looks good in general, just a minor comment

File: Cargo.toml (hunk 1 above)
Comment (plaes) [reply Cargo.toml]: Fixed

**Review by blackghost1987 — APPROVED** [review]

**Comment by blackghost1987** [comment]
Let me know if you need a Crates.io release
